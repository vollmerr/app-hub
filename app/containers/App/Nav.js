import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { Nav as OfficeNav } from 'office-ui-fabric-react/lib/Nav';

import toJS from '../../hocs/toJS';
import theme from '../../utils/theme';

import * as selectors from '../AppHub/selectors';


export const Wrapper = styled.div`
  display: ${(props) => props.isMobile ? 'none' : 'block'};
  flex: 0 0 ${theme.app.navWidth}px;

  .ms-FocusZone {
    height: calc(100vh - ${theme.hub.headerHeight}px);
    background: ${(props) => props.isMobile ? theme.neutralPrimary : theme.neutralLight};

    li > div {
      background: ${(props) => props.isMobile ? theme.neutralPrimary : theme.neutralLight};
      button {
        text-decoration: none;
        display: block;
        color: ${(props) => props.isMobile ? theme.white : 'inherit'};
        &:hover,
        &:focus,
        &:active {
          color: ${(props) => props.isMobile ? theme.neutralDark : 'inherit'};
          background: ${(props) => props.isMobile ? theme.neutralLighter : theme.neutralTertiaryAlt};
        }
      }
      &:hover button,
      &.is-selected button {
        background: ${(props) => props.isMobile ? theme.neutralLighter : theme.neutralTertiaryAlt};
      }
    }
  }
`;


// https://github.com/OfficeDev/office-ui-fabric-react/issues/915
export class Nav extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      selectedKey: null,
      routes: [],
    };
  }

  componentDidMount() {
    const { app, history } = this.props;
    // initalize to exisiting location
    this.getRoutes(app.routes, history.location);
    // listen for changes on history, updated selected on change
    /* istanbul ignore next */
    this.history = history.listen((location) => {
      this.getRoutes(this.props.app.routes, location); // use this.props to get new reference
    });
  }

  componentWillReceiveProps(nextProps) {
    // when mounted routes do not exist, and component is PureCOmponent must check here for route change
    if (nextProps.app.routes !== this.props.app.routes) {
      this.getRoutes(nextProps.app.routes, this.props.history.location);
    }
    return nextProps !== this.props;
  }

  componentWillUnmount() {
    // unlisten from history when unmounted (so not trying to update state on unmounted component)
    this.history();
  }

  /**
   * Maps the routes to only those that the user has role for
   *
   * @param {object} allRoutes  - routes to get map
   * @param {object} location   - user location info
   */
  getRoutes = (allRoutes, location) => {
    const { user } = this.props;
    const routes = allRoutes.filter((route) => {
      // ignore hidden routes
      if (route.hidden) {
        return false;
      }
      // no roles on route, keep on list of routes
      if (!route.roles) {
        return true;
      }
      // keep if user has correct role
      if (route.roles.some((role) => user.roles.includes(role))) {
        return true;
      }
      return false;
    });

    this.setState({ routes });
    this.getSelectedKey(routes, location);
  }

  /**
   * Handles getting the key of the selected nav item
   *
   * @param {object} routes     - routes to get selected key from (immutable)
   * @param {object} location   - user location info
   */
  getSelectedKey = (routes, location) => {
    const index = routes.findIndex((route) => route.path === location.pathname);
    // valid route
    if (index > -1) {
      this.setState({
        selectedKey: routes[index].key,
      });
    }
  }

  /**
   * Handles clicking on a link in the nav
   *
   * @param {event} event     - click event occuring
   * @param {object} element  - element being clicked
   */
  handleClickLink = (event, element) => {
    event.preventDefault();
    const { onClick, history } = this.props;

    if (onClick) {
      onClick(event, element);
    }

    const path = element.path || element.href;
    history.push(path);
  }

  render() {
    const { view } = this.props;
    const { selectedKey, routes } = this.state;

    return (
      <Wrapper isMobile={view.isMobile}>
        <OfficeNav
          selectedKey={selectedKey}
          groups={[{ links: routes }]}
          onLinkClick={this.handleClickLink.bind(this)} // eslint-disable-line
        />
      </Wrapper>
    );
  }
}


const { func, object } = PropTypes;

Nav.propTypes = {
  onClick: func,
  app: object.isRequired,
  view: object.isRequired,
  user: object.isRequired,
  history: object.isRequired,
};

const mapStateToProps = createStructuredSelector({
  app: selectors.getApp,
  user: selectors.getUser,
  view: selectors.getView,
});

const withConnect = connect(mapStateToProps);


export default compose(
  withRouter,
  withConnect,
  toJS,
)(Nav);
