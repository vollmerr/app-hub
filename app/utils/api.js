import { delay } from 'redux-saga';
import { call, put } from 'redux-saga/effects';
import decode from 'jwt-decode';
import 'whatwg-fetch';

import { authUserDone } from '../containers/AppHub/actions';


export const DEFAULT_EXPIRE = 1000;
export const TOKEN = 'id_token';


/**
 * Determines if should fetch based off a cache time
 *
 * @param {Date} date             - date of cache
 * @param {Number} cacheSeconds   - number of seconds to cache
 *
 * @return {Bool}                 - if should fetch
 */
export function shouldFetch(date, cacheSeconds = 350) {
  return !date || (new Date(date).getTime() - (cacheSeconds * 1000)) > new Date().getTime();
}


/**
 * Sets a token for authenication in localStorage
 *
 * @param {String} token  - token to set
 */
export function setToken(token) {
  localStorage.setItem(TOKEN, token);
}


/**
 * Gets the authenication token from localStorage
 * @return {String}     - authenication token
 */
export function getToken() {
  return localStorage.getItem(TOKEN);
}


/**
 * Clears the authenication token from localStorage
 */
export function clearToken() {
  localStorage.removeItem(TOKEN);
}


/**
 * Helper function to valdate that token exists and is not expired
 *
 * @param {object} token  - jwt token
 *
 * @return {bool}         - valid ? true : false
 */
export function validToken(token) {
  try {
    const { exp } = decode(token);
    const now = new Date().getTime().toString().substring(0, 10);

    return Number(exp) > Number(now);
  } catch (e) {
    return false;
  }
}


/**
 * Helper function to put jwt token into global redux store
 *
 * @param {String} token    - jwt token
 *
 * @return {Number}         - exipration time
 */
export function* putToken(token) {
  try {
    const {
      sub: sam,
      exp,
      iat,
      ...rest
    } = decode(token);

    const expire = (exp - iat) * 1000; // convert from ms to s

    yield put(authUserDone({ sam, expire, ...rest }));
    return expire;
  } catch (error) {
    throw new Error(error);
  }
}


/**
 * Authenticates a user
 * Checks local storage for token first
 *
 * @param {Object} action   - redux action to dispatch (must have 'type')
 * @param {Number} maxTries - max tries to re attempt authenicating on fail
 */
export function* authenticate(action, maxTries = 3) {
  let expire = DEFAULT_EXPIRE; // try again in 1s by default
  let tries = 0;
  let error = new Error('Failed to authenticate');
  while (tries < maxTries) { // eslint-disable-line
    const token = getToken();
    // if valid local token use that, otherwise get new token
    if (validToken(token)) {
      // update user in global appHub state
      expire = yield call(putToken, token);
      // reset max tries
      tries = 0;
    } else {
      try {
        const options = {
          method: 'get',
          credentials: 'include',
        };
        const { id_token } = yield call(request, API.JWT, options); // eslint-disable-line
        // set into local storage for future authentication caching
        setToken(id_token);
        // update user in global appHub state
        expire = yield call(putToken, id_token);
        // reset max tries
        tries = 0;
      } catch (err) {
        // increment max tries
        tries += 1;
        error = err;
      }
      // max number of tires, set error
      if (tries === maxTries) {
        yield put(authUserDone(error));
      }
    }
    // authenicate again when token has expired
    yield call(delay, expire);
  }
}


/**
 * Parses the JSON returned by a network request
 *
 * @param  {object} response A response from a network request
 *
 * @return {object}          The parsed JSON from the request
 */
export function parseJSON(response) {
  if (response.status === 204 || response.status === 205) {
    return null;
  }
  return response.json();
}


/**
 * Checks if a network request came back fine, and throws an error if not
 *
 * @param  {object} response   A response from a network request
 *
 * @return {object|undefined} Returns either the response, or throws an error
 */
export function checkStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  }

  const error = new Error(response.statusText);
  error.response = response;
  throw error;
}


/**
 * Requests a URL, returning a promise
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response data
 */
export function request(url, options) {
  return fetch(url, options)
    .then(checkStatus)
    .then(parseJSON);
}


/**
 * Requests a URL with jwt token in header for authorization, returning a promise
 * Waits until jwt token is set in localStorage before making call
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response data
 */
export function requestWithToken(url, options = {}) {
  let timer = null;

  const requestIfToken = () => { // eslint-disable-line
    // get users jwt token form storage else they dont have one
    const token = localStorage.getItem('id_token') || null;
    // valid token, call api
    if (validToken(token)) {
      clearTimeout(timer);
      // config settings to send api
      const newOptions = {
        method: options.method || 'get',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${token}`,
        },
        ...options,
      };
      // append optional body to request (POSTs)
      if (options.body) {
        newOptions.body = JSON.stringify(options.body);
      }

      return request(url, newOptions);
    }
    // wait again until token set
    timer = setTimeout(requestIfToken, 100);
  };

  return requestIfToken();
}
