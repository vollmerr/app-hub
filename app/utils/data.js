/**
 * Formats an item object to the correct format
 * for rendering
 *
 * @param {array} item      - item to format
 * @param {string} name     - internal field name for lookup
 * @param {object} field    - field format info
 * @param {object} enums    - display names for enums
 *
 * @return {string}         - formatted item
 */
export function formatItem(item, name, field, enums = {}) {
  let content = item[name];
  // convert all data to array for mapping over
  if (!Array.isArray(content)) {
    content = [content];
  }
  // go through each item, mapping to the correct format
  content = content.map((x) => {
    // is enum mapping
    if (enums[name]) {
      return enums[name][x];
    }
    // is date
    if (field.data && field.data.type === 'DATE') {
      return isNaN(Date.parse(x)) ? '' : new Date(x).toISOString().substr(0, 10);
    }
    return x;
  });
  // join them back as string seperated by commas
  return content.join(', ');
}

/**
 * Formats an array of item objects to the correct format
 * for rendering
 *
 * @param {array} items     - items to format
 * @param {object} fields   - lookup table of fields
 * @param {object} enums    - display names for enums
 *
 * @return {array}          - formatted items as objects of strings
 */
export function formatItems(items, fields, enums) {
  const formattedItems = [];

  items.forEach((item, i) => {
    formattedItems[i] = {};
    Object.keys(item).forEach((k) => {
      formattedItems[i][k] = formatItem(item, k, fields[k], enums);
    });
  });

  return formattedItems;
}

/**
 * Maps a object to the format Office-UI-Fabric-React DetailsList`s expect
 *
 * @param {object} obj    - object to map
 * @param {array} include - keys to include, if empty includes all
 * @param {array} exclude - keys to exclude
 *
 * @return {object}       - mapped object to use in List
 */
export function mapToColumns(obj, include = [], exclude = []) {
  const keys = include.length ?
    include :
    Object.keys(obj);

  return keys
    .filter((key) => !exclude.includes(key))
    .map((key) => ({
      key: obj[key].name,
      name: obj[key].label,
      fieldName: obj[key].name,
      isResizable: true,
      ariaLabel: obj[key].ariaLabel,
      minWidth: obj[key].minWidth || 100,
      maxWidth: obj[key].maxWidth || 200,
      data: obj[key].data || {},
    }));
}
