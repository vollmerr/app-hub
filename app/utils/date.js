export const dateWithoutTime = (date) => {
  if (date && !isNaN(new Date(date))) {
    const d = new Date(date);
    d.setHours(0, 0, 0, 0);
    return d;
  }
  return undefined;
};
