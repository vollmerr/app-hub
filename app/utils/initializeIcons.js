import { registerIcons } from '@uifabric/styling/lib/index';
import appHubIcons from 'images/office365icons.woff';

// http://o365icons.cloudapp.net/
const initializeIcons = () => {
  registerIcons({
    style: {
      MozOsxFontSmoothing: 'grayscale',
      WebkitFontSmoothing: 'antialiased',
      fontStyle: 'normal',
      fontWeight: 'normal',
      speak: 'none',
    },
    fontFace: {
      fontFamily: `"appHubIcons"`, // eslint-disable-line
      src: `url(${appHubIcons}) format('woff')`,
    },
    icons: {
      // NAV
      helpMenu: '\uE006',
      alertsMenu: '\uE010',
      appMenu: '\uE020',
      settingsMenu: '\uE035',
      appHubMenu: '\uE303',
      // __APPS__
      ED: '\uE417',
      Loading: '\uF8FD',
      Demo: '\uF8FD',
      // SPA: '\uE22E',
      Apphub: '\uE300',
      COLD: '\uE408',
      BARS: '\uE259',
      // requried by office-ui
      chevrondown: '\uE088',
      chevronup: '\uE087',
      Search: '\uE039',
      Clear: '\uE353',
      CircleRing: '\uF8FE',
      StatusCircleCheckmark: '\uF8FF',
      sortup: '\uF8FB',
      sortdown: '\uF8FC',
      calendar: '\uE38E',
      up: '\uE27A',
      down: '\uE27B',
      checkmark: '\uF8FA',
    },
  });
};

export default initializeIcons;
