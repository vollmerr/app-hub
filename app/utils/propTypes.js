import PropTypes from 'prop-types';

const { string, bool, shape, func, arrayOf } = PropTypes;

export const metaDataProp = shape({
  title: string,
  desc: string,
  keywords: string,
});

export const routeProp = shape({
  key: string,
  href: string,
  path: string,
  name: string,
});
export const routesProp = arrayOf(routeProp);

// REDUX-FORM
export const inputProp = (value) => shape({
  value,
  onChange: func,
  onBlur: func,
  onFocus: func,
});

export const metaProp = shape({
  error: string,
  touched: bool,
});

export const optionProp = shape({
  key: string,
  text: string,
});
export const optionsProp = arrayOf(optionProp);

