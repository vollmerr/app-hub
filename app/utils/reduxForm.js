export const mapOnBlur = (onBlur, input) => (event, newValue, previousValue) => {
  if (onBlur) {
    onBlur(event, newValue, previousValue);
  }
  input.onBlur(event, newValue, previousValue);
};

export const mapOnFocus = (onFocus, input) => (event) => {
  if (onFocus) {
    onFocus(event);
  }
  input.onFocus(event);
};

export const mapOnChange = (onChange, input) => (event, newValue, previousValue) => {
  if (onChange) {
    onChange(event, newValue, previousValue);
  }
  input.onChange(event, newValue, previousValue);
};
