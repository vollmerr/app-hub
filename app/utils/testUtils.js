import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

/**
 * Tests a styled-component for basic tests
 * such as rending correctly and adopting attributes
 * @param {Object} Component    - React component under test
 * @param {Object} extend       - React component the component under test extends /  inherits from
 */
export function testStyledComponent(Component, extend = null) {
  describe(`<${Component.displayName} />`, () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Component />);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it(`should render an <${Component.target}> tag`, () => {
      expect(wrapper.type()).toEqual(Component.target);
    });

    it('should have a className attribute', () => {
      expect(wrapper.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      wrapper.setProps({ id });
      expect(wrapper.prop('id')).toEqual(id);
    });

    if (extend) {
      it(`should extend ${extend.name}`, () => {
        expect(wrapper.find(extend).exists()).toEqual(true);
      });
    } else {
      it('should not adopt an invalid attribute', () => {
        const attribute = 'test';
        wrapper.setProps({ attribute });
        expect(wrapper.prop('attribute')).toBeUndefined();
      });
    }
  });
}

/**
 * Tests a components for basic mapDispatchToProps information
 * @param {func} mapDispatchToProps   - components mapDispatchToProps function
 * @param {object} actions            - functions in mapDispatchToProps to test
 */
export function testMapDispatchToProps(mapDispatchToProps, actions) {
  const onAction = (action) => (
    `on${action.charAt(0).toUpperCase()}${action.slice(1)}`
  );

  let dispatch;
  let dispatchProps;
  beforeEach(() => {
    dispatch = jest.fn();
    dispatchProps = mapDispatchToProps(dispatch);
  });

  Object.keys(actions).forEach((action) => {
    describe(onAction(action), () => {
      it('should be injected', () => {
        expect(dispatchProps[onAction(action)]).toBeDefined();
      });

      it('should dispatch when called', () => {
        dispatchProps[onAction(action)]();
        expect(dispatch).toHaveBeenCalled();
      });
    });
  });
}
