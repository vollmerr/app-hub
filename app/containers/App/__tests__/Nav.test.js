import React from 'react';
import { shallow, mount } from 'enzyme';
import { Nav as OfficeNav } from 'office-ui-fabric-react/lib/Nav';

import { testStyledComponent } from '../../../utils/testUtils';

import { Nav, Items } from '../Nav';


testStyledComponent(Items, OfficeNav);


describe('<Items />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Items isMobile={false} />);
  });

  it('should render correctly in desktop view', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly in desktop view', () => {
    wrapper.setProps({ isMobile: true });
    expect(wrapper).toMatchSnapshot();
  });
});


const props = {
  onClick: jest.fn(),
  app: {
    routes: [
      { name: 'routePublic', path: '/routePublic', key: 'key1' },
      { name: 'routeUnauth', path: '/routeUnauth', key: 'key2', roles: ['otherRole'] },
      { name: 'routeAuth', path: '/routeAuth', key: 'key3', roles: ['role1'] },
    ],
  },
  view: {
    isMobile: false,
  },
  user: {
    roles: ['role1'],
  },
  history: {
    location: {
      pathname: '/routePublic',
    },
    listen: jest.fn(),
    push: jest.fn(),
  },
};


describe('<Nav />', () => {
  let wrapper;
  let instance;
  let event;
  let element;
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(<Nav {...props} />);
    instance = wrapper.instance();
    event = { preventDefault: jest.fn() };
    element = { path: '/test' };
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });


  describe('componentDidMount', () => {
    beforeEach(() => {
      instance.getRoutes = jest.fn();
      const history = { ...props.history, listen: jest.fn(() => instance.getRoutes) };
      wrapper.setProps({ history });
      instance.componentDidMount();
    });

    it('should update the routes to only those the user has access to', () => {
      expect(instance.getRoutes).toHaveBeenCalledWith(props.app.routes, props.history.location);
    });

    it('should bind the history listener to `this.history`', () => {
      expect(instance.history).toBeDefined();
    });
  });


  describe('componentWillReceiveProps', () => {
    // not resetting between tests to preserve state
    const mounted = mount(<Nav {...props} />);
    const newProps = { ...props, app: { routes: [{ name: 'new route' }, ...props.app.routes] } };
    const inst = mounted.instance();
    const render = jest.spyOn(inst, 'render');
    inst.getSelectedKey = jest.fn();

    it('should re-render and update the selectedKey when new `appRoutes` passed', () => {
      mounted.setProps({ ...newProps });
      expect(inst.getSelectedKey).toHaveBeenCalled();
      expect(render).toHaveBeenCalled();
    });

    it('should not update when same new props passed', () => {
      jest.resetAllMocks();
      mounted.setProps({ ...newProps });
      expect(inst.getSelectedKey).not.toHaveBeenCalled();
      expect(render).not.toHaveBeenCalled();
    });

    it('should re-render but not update the selectedKey when new props but same `app.routes` passed', () => {
      jest.resetAllMocks();
      newProps.isMobile = true;
      mounted.setProps({ ...newProps });
      expect(inst.getSelectedKey).not.toHaveBeenCalled();
      expect(render).toHaveBeenCalled();
    });
  });


  describe('componentWillUnmount', () => {
    it('should unsubscribe from history', () => {
      instance.history = jest.fn();
      wrapper.unmount();
      expect(instance.history).toHaveBeenCalled();
    });
  });


  describe('getRoutes', () => {
    let filteredRoutes;
    beforeEach(() => {
      instance.getSelectedKey = jest.fn();
      filteredRoutes = [props.app.routes[0], props.app.routes[2]]; // routes user is authed for
    });

    it('should update the `selectedKey`', () => {
      instance.getRoutes(props.app.routes, history.location);
      expect(instance.getSelectedKey).toHaveBeenCalledWith(filteredRoutes, history.location);
    });

    it('should update the `routes` in state', () => {
      instance.getRoutes(props.app.routes, history.location);
      expect(wrapper.state('routes')).toEqual(filteredRoutes);
    });
  });


  describe('getSelectedKey', () => {
    it('should exist', () => {
      expect(instance.getSelectedKey).toBeDefined();
    });

    it('should set the selectedKey if path found', () => {
      const location = { pathname: props.app.routes[2].path };
      instance.getSelectedKey(props.app.routes, location);
      expect(wrapper.state('selectedKey')).toEqual(props.app.routes[2].key);
    });

    it('should set not the selectedKey if path not found', () => {
      const location = { pathname: '' };
      instance.getSelectedKey(props.app.routes, location);
      expect(wrapper.state('selectedKey')).toEqual(props.app.routes[0].key);
    });
  });


  describe('handleClickLink', () => {
    it('should exist', () => {
      expect(instance.handleClickLink).toBeDefined();
    });

    it('should call onClick if passed one', () => {
      instance.handleClickLink(event, element);
      expect(props.onClick).toHaveBeenCalledWith(event, element);
    });

    it('should not call onClick if not passed one', () => {
      wrapper.setProps({ onClick: undefined });
      instance.handleClickLink(event, element);
      expect(props.onClick).not.toHaveBeenCalled();
    });

    it('should handle routing for internal links (path)', () => {
      instance.handleClickLink(event, element);
      expect(props.history.push).toHaveBeenCalledWith(element.path);
    });

    it('should handle routing for external links (href)', () => {
      element = { href: 'http://www.google.com' };
      instance.handleClickLink(event, element);
      expect(props.history.push).toHaveBeenCalledWith(element.href);
    });
  });
});
