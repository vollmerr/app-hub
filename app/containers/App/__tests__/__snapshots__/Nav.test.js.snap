// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> render should render correctly 1`] = `
<Nav__Wrapper
  isMobile={false}
>
  <Nav
    groups={
      Array [
        Object {
          "links": Array [
            Object {
              "key": "key1",
              "name": "routePublic",
              "path": "/routePublic",
            },
            Object {
              "key": "key3",
              "name": "routeAuth",
              "path": "/routeAuth",
              "roles": Array [
                "role1",
              ],
            },
          ],
        },
      ]
    }
    onLinkClick={[Function]}
    selectedKey="key1"
  />
</Nav__Wrapper>
`;

exports[`<Nav__Wrapper /> should render correctly 1`] = `
.c0 {
  -webkit-flex: 0 0 225px;
  -ms-flex: 0 0 225px;
  flex: 0 0 225px;
}

.c0 .ms-FocusZone {
  height: calc(100vh - 50px);
  background: #eaeaea;
}

.c0 .ms-FocusZone li > div {
  background: #eaeaea;
}

.c0 .ms-FocusZone li > div button {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  color: inherit;
}

.c0 .ms-FocusZone li > div button:hover,
.c0 .ms-FocusZone li > div button:focus,
.c0 .ms-FocusZone li > div button:active {
  color: inherit;
  background: #c8c8c8;
}

.c0 .ms-FocusZone li > div:hover button,
.c0 .ms-FocusZone li > div.is-selected button {
  background: #c8c8c8;
}

<div
  className="c0"
/>
`;

exports[`<Wrapper /> should render correctly in desktop view 1`] = `
.c0 {
  -webkit-flex: 0 0 225px;
  -ms-flex: 0 0 225px;
  flex: 0 0 225px;
}

.c0 .ms-FocusZone {
  height: calc(100vh - 50px);
  background: #eaeaea;
}

.c0 .ms-FocusZone li > div {
  background: #eaeaea;
}

.c0 .ms-FocusZone li > div button {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  color: inherit;
}

.c0 .ms-FocusZone li > div button:hover,
.c0 .ms-FocusZone li > div button:focus,
.c0 .ms-FocusZone li > div button:active {
  color: inherit;
  background: #c8c8c8;
}

.c0 .ms-FocusZone li > div:hover button,
.c0 .ms-FocusZone li > div.is-selected button {
  background: #c8c8c8;
}

<div
  className="c0"
/>
`;

exports[`<Wrapper /> should render correctly in desktop view 2`] = `
.c0 {
  -webkit-flex: 0 0 225px;
  -ms-flex: 0 0 225px;
  flex: 0 0 225px;
}

.c0 .ms-FocusZone {
  height: calc(100vh - 50px);
  background: #333333;
}

.c0 .ms-FocusZone li > div {
  background: #333333;
}

.c0 .ms-FocusZone li > div button {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  color: #ffffff;
}

.c0 .ms-FocusZone li > div button:hover,
.c0 .ms-FocusZone li > div button:focus,
.c0 .ms-FocusZone li > div button:active {
  color: #212121;
  background: #f4f4f4;
}

.c0 .ms-FocusZone li > div:hover button,
.c0 .ms-FocusZone li > div.is-selected button {
  background: #f4f4f4;
}

<div
  className="c0"
/>
`;
